### PR TITLE
Implement `lapin_futures::channel::Channel::basic_qos` method

### PR DESCRIPTION
**Pull request by swizard0**

### CODE REVIEW

File: futures/src/lib.rs [hunk 1]
```diff
@@ -143,9 +143,12 @@ extern crate lapin_async;
 extern crate nom;
 extern crate tokio_io;
 extern crate tokio_timer;
+#[cfg(test)] extern crate tokio_core;
```
Comment (Keruspe) [inline futures/src/lib.rs]: please don't do that and use the futures/tests directory instead

File: futures/src/tests.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+use futures::future::Future;
+use futures::Stream;
+use tokio_core::reactor::Core;
+use tokio_core::net::TcpStream;
+use super::client::{self, ConnectionOptions};
+use super::types::FieldTable;
+
+#[test]
+fn qos_and_queue() {
```
Comment (Keruspe) [inline futures/src/tests.rs]: I don't think adding a whole new test is relevant, just hook up the new method in the existing test

File: futures/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "lapin-futures"
-version = "0.10.0"
+version = "0.10.1"
```
Comment (Keruspe) [inline futures/Cargo.toml]: Please don't update the version number, it will get updated when we release a new version

File: futures/src/tests.rs [hunk 4]
```diff
@@ -19,18 +19,22 @@ fn qos_and_queue() {
...
-                        "",
-                        "qos_and_queue",
-                        b"hello from lapin_futures",
-                        &Default::default(),
-                        Default::default())
-                        .map(move |_| channel)
-                })
-                .and_then(move |channel| {
-                    channel.queue_delete("qos_and_queue", &Default::default())
+            channel.basic_qos(&BasicQosOption { prefetch_count: 16, ..Default::default() })
```
Comment (Keruspe) [inline futures/src/tests.rs]: please move that to the test in the tests directory

Comment (Keruspe) [inline futures/src/tests.rs]: A whole new test doesn't seem necessary, just adding the call in the existing test should be enough

File: futures/src/lib.rs [hunk 5]
```diff
@@ -144,6 +144,7 @@ extern crate nom;
 extern crate tokio_io;
 extern crate tokio_timer;
 
+
```
Comment (Keruspe) [inline futures/src/lib.rs]: Why the empty line?

**Comment by swizard0** [comment]
> Can you squash the commits please?

Currently trying to :) Sorry, I'm not very familiar with git

**Comment by Keruspe** [comment]
Thanks !

**Comment by samueltardieu** [comment]
Will there be a release soon which includes this?

**Pull request merged**